src/filtrering/filtrering-mine-filter/dragAndDrop/drag-and-drop.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useState, useRef, useEffect, useCallback} from 'react';
+import './drag-and-drop.less';
+import {MineFilter, lagreSorteringForFilter} from '../../../ducks/mine-filter';
+import DragAndDropContainer from './drag-and-drop-container';
+import NyMineFilterRad from '../ny_mine-filter-rad';
+import {useDispatch} from 'react-redux';
+
+export interface DragAndDropProps {
+    stateFilterOrder: MineFilter[];
+    filtergruppe: string;
+    isDraggable: boolean;
+    setisDraggable: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+function DragAndDrop({stateFilterOrder, filtergruppe, isDraggable, setisDraggable}: DragAndDropProps) {
+    const [dragAndDropOrder, setDragAndDropOrder] = useState([...stateFilterOrder]);
+    const dispatch = useDispatch();
+
+    const lagreRekkefølge = useCallback(() => {
+        const idAndPriorities = dragAndDropOrder.map((filter, idx) => ({
+            sortOrder: idx,
+            filterId: filter.filterId
+        }));
+        if (harEndretRekkefølge(dragAndDropOrder, stateFilterOrder)) {
+            dispatch(lagreSorteringForFilter(idAndPriorities));
+        }
+        setisDraggable(false);
+    }, [dragAndDropOrder, stateFilterOrder, setisDraggable, dispatch]);
+
+    const onUnmountRef = useRef(() => {
```

Comment:
why its necessary to use useRef? This logic is bit confusing.

Reply:
It done so that the onUnmountRef never causes react to re-render. Then it is possible to only run the onUnmount code when the DragAndDropContainer is unMounted. But yes I also think that this is a bit confusing, but I dont know a better way that does not causes many or infinit re-renders. I am open to discuss if you want to, but it may be easier over the phone?

Reply:
Let me investigate a bit this code..

Reply:
Is it the point to save sort order after sort-toggle is off?

Reply:
@stianStensli I pushed some small changes, can you pls verify its working still ok.